download.py

```diff
@@ -40,91 +48,112 @@
+        header_row = entries[0]
+        dataset = entries[1:]
+        return header_row, dataset
+
+def download_entry_audio(entry, raw_audio_dir, audio_dir, use_cache=True, verbose=False):
     """
     Downloads an audio recording given its entry in a Tarteel dataset csv.
     """
-    surah_num = int(row[0])
-    ayah_num = int(row[1])
-    url = row[2]
-    parsed_url = urlparse(url)
-    wav_filename = os.path.basename(parsed_url.path)
-    local_download_path = recording_utils.get_path_to_recording(local_download_dir, surah_num, ayah_num, wav_filename)
-
-    # Download and save the audio recording to the given path.
-    try:
-        with requests.Session() as s:
-            if args.verbose:
-                print("Downloading", url, "to", local_download_path, ".")
-            download = s.get(url)
-            dirname = os.path.dirname(local_download_path)
-            if not os.path.exists(dirname):
-                os.makedirs(dirname)
-            file = open(local_download_path, "wb")
-            file.write(download.content)
-            file.close()
-
-    except:
-        # If the download fails, print an error and exit the function.
-        print("Audio file", local_download_path, "could not be opened.")
-        return
-
-    # Check if the wave header is valid and if so, get the desired info. This deletes files with invalid headers.
-    has_valid_header, wav_bytes, sample_rate_hz, num_channels = recording_utils.open_recording(local_download_path)
-
-    if has_valid_header:
-        # Note: webrtcVAD does not currently support 44.1MHz, so we have no way of checking those files for empty audio.
-        recording_has_speech = recording_utils.has_speech(
-            wav_bytes, 
-            sample_rate_hz, 
-            num_channels, 
-            non_speech_threshold_fraction=DEFAULT_NON_SPEECH_THRESHOLD_FRACTION, 
-            verbose=args.verbose)
-
-        if args.vad_check and sample_rate_hz in WEBRTCVAD_SUPPORTED_SAMPLE_RATES_HZ and not recording_has_speech:
-            print("Audio file", local_download_path, "does not have speech according to VAD. Removing.")
-            os.remove(local_download_path)
-
-    if sample_rate_hz not in WEBRTCVAD_SUPPORTED_SAMPLE_RATES_HZ+[44100] and sample_rate_hz != None:
-        # If unsupported sample frequency, remove the file.
-        print("File ", local_download_path, " has an unsupported sample frequency %d. Removing." % sample_rate_hz)
-        if os.path.exists(local_download_path):
-            os.remove(local_download_path)
+    surah_num, ayah_num, url, _, _, _, _, _, _, _, _ = entry
+
+    # Convert data to correct type from string.
+    surah_num = int(surah_num)
+    ayah_num = int(ayah_num)
+
+    # Ensure that the directories that the recording will be saved to exist.
+    downloaded_ayah_audio_dir = file_utils.prepare_ayah_directory(raw_audio_dir, surah_num, ayah_num)
+    ayah_audio_dir = file_utils.prepare_ayah_directory(raw_audio_dir, surah_num, ayah_num)
```

Comment:
`downloaded_ayah_audio_dir` and `ayah_audio_dir` seem to be the same thing?

Reply:
Fixed.